src/sas/qtgui/Perspectives/Fitting/ConstraintWidget.py

```diff
@@ -493,12 +493,46 @@ def fitComplete(self, result):
             self.parent.communicate.statusBarUpdateSignal.emit(msg)
             return
 
+        # Get the results list
+        results = result[0][0]
+        # Warn the user if fitting has been unsuccessful
+        if not results[0].success:
+            msg = results[0].mesg
+            # if the exception is a NameError, warn the user of which constraint is faulty
+            if type(msg[0]) == NameError:
+                # get the exception in the traceback
+                name = str(msg[0]).split("'")[1]
+                trace = results[0].trace
+                # Hop to the last trace to find the original exception
+                while trace.tb_next:
+                       trace = trace.tb_next
+                # get the line of the faulty constraint in the code
+                lineno = trace.tb_lineno
+                # get the code that assigns the constraints
+                constraint_code = trace.tb_frame.f_code.co_filename.split('\n')
+                # get the incriminated constraint from the code (see expression.py for details)
+                line = lineno - 2 - (len(constraint_code)-6)/2
+                faulty_constraint = constraint_code[int(line)]
+                constraint_no = self.findConstraintInTable(faulty_constraint)
+                name_error = self.findNameErrorInConstraint(self.tblConstraints.item(constraint_no, 0).text(),
+                                                            faulty_constraint,
+                                                            name)
+                # warn the user of which constraint is failing and highlight the error
+                error_text = 'Fit failed because constraint <b>%s</b> is inconsistent:<br>%s'%(
+                    constraint_no + 1,
+                    self.tblConstraints.item(constraint_no, 0).text().replace(name_error, "<b>"+name_error+"</b>"))
+            else:
+                error_text = 'Fit failed'
+            QtWidgets.QMessageBox.warning(self,
+                                          'Warning',
+                                          error_text,
+                                          QtWidgets.QMessageBox.Ok)
```

Comment:
Good point. Not a very good idea to change the workflow. I believe that batch fitting would cause a separate box to pop up for each fit, which is a problem. I changed the behaviour so it show the message in the status bar